Resources/views/frontend/_public/src/js/applepay.js

```diff
@@ -0,0 +1,49 @@
+(function ($) {
+    "use strict";
+
+    $(document).ready(function() {
+        hideApplePayIfNotAllowed();
+    });
+
+    $(document).ajaxComplete(function() {
+        hideApplePayIfNotAllowed();
+    });
+
+    /**
+     * This function is used to hide Apple Pay if the visitor can't use it.
+     * Please keep in mind, this is Apple Pay and not Apple Pay Direct.
+     * So it just hides Apple Pay on the payment selection page.
```

Comment:
wouldn't it be better to run all this only on checkout/payment (don't know the exact controller action related css class of body tag but there might be a specific one to distinguish)? otherwise that js code will be fired on any page's load and even on any ajaxComplete event... please think about that

Reply:
yes in refactoring, for now the prio is to revert that functionality to have it back